src/LayerEditor/ui/panels/regions_panel.py

```diff
@@ -32,25 +33,15 @@ class LayerHeads(QtCore.QObject):
     region_list_changed = QtCore.pyqtSignal()
 
 
-    def __init__(self, layers_geometry):
+    def __init__(self, block):
```

Comment:
BlockLayereads could be completely replaced by suitable getters and modification methods of the BlockModel.

Reply:
It was replaced by modified LEModel since LEModel contains information about currently active block. This solved several discussed problems. Done.